cmake/try_compile/us_have_regex.cpp

```diff
@@ -1,5 +1,7 @@
 #include <regex>
-int main() {
+int
```

Comment:
For the function definitions, my preference would be something like
```cpp
int main()
{
```
instead of 
```cpp
int
main()
{
```

( I mean if the consensus is we stick to the Mozilla default style i.e. option 2 going forward, I am fine but my preference is #1 :-).  )

I believe the configuration options are `AlwaysBreakAfterDefinitionReturnType`  and `AlwaysBreakAfterReturnType`

Reply:
Okay. I am also going to add indentation for pre-processor directives. Also, we can easily change the style later if we find we are not happy with something.